src/Analytics/BasicProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Message\Mothership\CMS\Analytics;
+
+/**
+ * @author Samuel Trangmar-Keates
+ *
+ * A basic provider class which is constructed with the reference, params and names
+ * required. Useful for basic providers.
+ *
+ * If there is a more complecated provider, then AnalyticsProviderInterface should
+ * be implemented
+ */
+class BasicProvider implements AnalyticsProvidorInterface
+{
+	/**
+	 * @var string
+	 */
+	private $_name;
+
+	/**
+	 * @var string
+	 */
+	private $_viewRef;
+
+	/**
+	 * @var array
+	 */
+	private $_params;
+
+	/**
+	 * Construct the object with the correct fields.
+	 * 
+	 * @param string $name    The object identifier
+	 * @param string $viewRef The view reference
+	 * @param array  $params  The parameters to pass to the view
+	 */
+	public function __construct($name, $viewRef, $params = [])
```

Comment:
I'm not sure I agree it makes them less flexible. You type hint with the interface anyway, but also allows type hinting on the individual provider classes, and it makes them easier to identify if dumping or whatever. In fact I think this way is less flexible because you lose control of the constructor. Also on that note, you need to add an extra parameter to the constructor too, so I really do think it should be the other way. You are still using this class because you extend it:

```php
class GoogleAnalyticsProvider extends BaseProvider
{
    public function __construct($viewRef = 'Some:Default::view:reference', $params = [])
    {
        parent::__construct('google-analytics', $viewRef, $params);
    }
}
```